From 3bc29a941f235f432e583a7533b4ac1d97597e05 Mon Sep 17 00:00:00 2001
From: Jannick <thirdedition@gmx.net>
Date: Sun, 7 Jun 2020 11:30:00 +0200
Subject: [PATCH] foreign.c: fix newline on Windows

Windows specifics: '\n' is printed as '\r\n' to a stream, unless the
stream is opened in binary mode.  Thus '\r\n' is converted to '\r\r\n'
which causes additional newlines.  This needs to be amended and appears
in igraph's pajek tests.

See commits 73e66786, 8b14a2571.

* src/foreign.c: replace '\x0d\0a' ('\r\n') by '\x0a' ('\n').
---
 src/foreign.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/foreign.c b/src/foreign.c
index 6019b93..282d0db 100644
--- a/src/foreign.c
+++ b/src/foreign.c
@@ -2091,7 +2091,7 @@ int igraph_write_graph_pajek(const igraph_t *graph, FILE *outstream) {
                               };
     const char *estrnames2[] = { "a", "p", "l", "lc", "c" };
 
-    const char *newline = "\x0d\x0a";
+    const char *newline = "\x0a";
 
     igraph_es_t es;
     igraph_eit_t eit;
-- 
2.27.0.windows.1

